### PR TITLE
[CCFPCM-0574] report trigger from reconciler

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[CCFPCM-0574](https://bcdevex.atlassian.net/browse/CCFPCM-0574)

Objective: 

- SNS set up

- trigger report from reconcilaition
- (locally you still have to call make reconcile but the report will generate automatically)

TODOS:

- dynamically set the Ministry

### CODE REVIEW

File: apps/backend/src/lambdas/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { parse } from 'date-fns';
+/*eslint-disable */
+export const parseSNSMessage = (parsedMessage: any) => {
+  const from_date = parse(
+    parsedMessage.Records[0].Sns.Message.period.from,
+    'yyyy-MM-dd',
+    new Date()
+  );
+  console.log(from_date);
```
Comment (fw-noel) [inline apps/backend/src/lambdas/helpers.ts]: Log

File: apps/backend/src/lambdas/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const to_date = parse(
+    parsedMessage.Records[0].Sns.Message.period.to,
```
Comment (fw-noel) [inline apps/backend/src/lambdas/helpers.ts:12]: Do we need to do some type checking around this function? e.g. at the start, check that parsedMessage.records?.length > 0, and that the necessary fields exist `Sns.Message.period.to`

File: apps/backend/src/lambdas/parser.ts [hunk 3]
```diff
@@ -1,18 +1,29 @@
...
+import { format, parse, subBusinessDays } from 'date-fns';
 import { ParseEvent } from './interface';
 import { AppModule } from '../app.module';
+import { Ministries } from '../constants';
 import { AppLogger } from '../logger/logger.service';
 import { ParseService } from '../parse/parse.service';
+import { SnsManagerService } from '../sns-manager/sns-manager.service';
 
 const API_URL = process.env.API_URL;
+const SNS_PARSER_RESULTS_TOPIC = process.env.SNS_PARSER_RESULTS_TOPIC;
```
Comment (fw-noel) [inline apps/backend/src/lambdas/parser.ts:12]: Do we need this in the `env.example`?

File: apps/backend/src/lambdas/parser.ts [hunk 4]
```diff
@@ -1,18 +1,29 @@
...
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export const handler = async (event: ParseEvent, _context?: Context) => {
   const app = await NestFactory.createApplicationContext(AppModule);
   const appLogger = app.get(AppLogger);
+  const snsService = app.get(SnsManagerService);
+  const parseService = app.get(ParseService);
+  const isLocal = process.env.RUNTIME_ENV === 'local';
 
+  const fiscalStartDate = isLocal
```
Comment (fw-noel) [inline apps/backend/src/lambdas/parser.ts:23]: Comment to explain this, and remove console log

File: apps/backend/src/lambdas/parser.ts (hunk 3 above)
Comment (chelsea-EYDS) [reply apps/backend/src/lambdas/parser.ts:12]: added

**Review by fw-noel — APPROVED** [review]

**Review by fw-noel — APPROVED** [review]

**Pull request merged**